src/main/java/com/mathworks/ci/MatlabBuilder.java

```diff
@@ -410,15 +413,16 @@ private synchronized int execMatlabCommand(Run<?, ?> build, FilePath workspace,
             throws IOException, InterruptedException {
         setEnv(build.getEnvironment(listener));
         final String testRunMode = this.getTestRunTypeList().getDescriptor().getId();
+        FilePath targetWorkspace = new FilePath(launcher.getChannel(), workspace.getRemote());
         
         // Copy MATLAB scratch file into the workspace only if Automatic option is selected.
         if (testRunMode.contains(AUTOMATIC_OPTION)) {
-            copyMatlabScratchFileInWorkspace(MATLAB_RUNNER_RESOURCE, MATLAB_RUNNER_TARGET_FILE,
-                    workspace, getClass().getClassLoader());
+            copyMatlabScratchFileInWorkspace(MATLAB_RUNNER_RESOURCE, MATLAB_RUNNER_TARGET_FILE, targetWorkspace);
         }
         ProcStarter matlabLauncher;
         try {
-            MatlabReleaseInfo rel = new MatlabReleaseInfo(getLocalMatlab());
+            FilePath nodeSpecificMatlabRoot = new FilePath(launcher.getChannel(),getLocalMatlab());
+            MatlabReleaseInfo rel = new MatlabReleaseInfo(nodeSpecificMatlabRoot);
             matlabLauncher = launcher.launch().pwd(workspace).envs(this.env);
             if (rel.verLessThan(BASE_MATLAB_VERSION_BATCH_SUPPORT)) {
```

Comment:
matlab command construction should be its own methods and should be able to work with a couple of inputs.